File: OSandDS/data structures/C++/homework2.cpp
#include <iostream>
#include <string>
using namespace std;

string find_answer(string message){
    string answers[5] = {
        "Sure", "Whoa, chill out!", "Calm down, I know what I am doing!", "Fine. Be that way!", "Whatever"
    };
    int lower_flag = 0;
    int question_flag = 0;
    
    if (message.find_first_not_of(' ') == string::npos || message == ""){
        return answers[3];
    }
    else{
        for (char i_char : message){
            if (isalpha(i_char) && tolower(i_char) == i_char){
                lower_flag = 1;
            }
            else if (i_char == '?'){
                question_flag = 1;
            }
        }
        
        if (lower_flag == 1 && question_flag == 0){
            return answers[4];
        }
        else if (lower_flag == 1 && question_flag == 1){
            return answers[0];
        }
        else if (lower_flag == 0 && question_flag == 0){
            return answers[1];
        }
        else if (lower_flag == 0 && question_flag == 1){
            return answers[2];
        }
    }
}

int main() {
    while (true){
        string text;
        cout << "Введите ваше сообщение: ";
        getline(cin, text);
        if (text == "goodbye"){
            cout << "Goodbye!" << endl;
            break;
        }
        else{
            cout << find_answer(text) << endl;
        }
    }
    return 0;
}

File: OSandDS/data structures/Python/homework2.py
# how are you ? = Sure
# FASFAFFS = Whoa, chill out!
# SFAFSAS? = Calm down, i know what iam doing!
#   = Fine.Be that way!
# остальные случаи = Whatever


def find_answer(massage):
    answers = {
        0: 'Sure',
        1: 'Whoa, chill out!',
        2: 'Calm down, i know what iam doing!',
        3: 'Fine.Be that way!',
        4: 'Whatever'
    }
    lower_flag = 0
    question_flag = 0
    if massage.isspace() or massage == '':
        return answers[3]
    else:
        for i_char in massage:
            if i_char.isalpha() and i_char == i_char.lower():
                lower_flag = 1
            elif i_char == '?':
                question_flag = 1
    if lower_flag == 1 and question_flag == 0:
        return answers[4]
    elif lower_flag == 1 and question_flag == 1:
        return answers[0]
    elif lower_flag == 0 and question_flag == 0:
        return answers[1]
    elif lower_flag == 0 and question_flag == 1:
        return answers[2]


while True:
    text = input("Введите ваше сообщение: ")
    if text.lower() == 'goodbay':
        print('Goodbay!')
        break
    else:
        print(find_answer(text))
